test/test_utils.py

```diff
@@ -259,6 +259,19 @@ def run_fn(input):
 
             self.assertEqual(grad_with_checkpointing, grad_no_checkpointing)
 
+    @unittest.skipIf(not HAS_CUDA, 'No CUDA')
+    def test_checkpoint_not_preserve_rng_state_and_without_reentrant(self):
+        inp = torch.randn(20000, device='cuda').requires_grad_()
+        layer = torch.nn.Dropout()
+
+        def run_fn(input):
+            return layer(input)
+
+        out = checkpoint(run_fn, inp, use_reentrant=False, preserve_rng_state=False)
+        out.sum().backward()
+        assert True
```

Comment:
No need for this.
In general, we shouldn't have plain asserts in the test suite (you want self.assertTrue()).
Here, you can simply leave a comment saying "Make sure that this runs without error".

Reply:
Yes that makes sense